lib/oidcstrategy.js

```diff
@@ -752,4 +577,494 @@ Strategy.prototype._shouldLoadUserProfile = function shouldLoadUserProfile(issue
   return done(null, !skip);
 };
 
+/**
+ * validate id_token, and pass the validated claims and the payload to callback
+ * if code (resp. access_token) is provided, we will validate the c_hash (resp at_hash) as well
+ *
+ * @param {String} id_token
+ * @param {Object} req
+ * @param {Function} callback
+ * @param {String} code (if you want to validate c_hash)
+ * @param {String} access_token (if you want o validate at_hash)
+ */
+Strategy.prototype._validateResponse = function validateResponse(id_token, code, access_token, req, callback) {
+  const self = this;
+
+  // decode id_token
+  const decoded = jws.decode(id_token);
+  if (decoded == null)
+    return self. fail(null, false, 'Invalid JWT token');
+  log.info('token decoded: ', decoded);
+
+  // get Pem Key
+  var PEMkey = null;
+  if (decoded.header.x5t) {
```

Comment:
I would reorder to check 'kid' first then 'x5t' as 'x5t' is an artifact of an older world and most likely will be dropped.